src/py21cmfast/io/h5.py

```diff
@@ -339,23 +350,51 @@ def _read_outputs(group: h5py.Group):
             stacklevel=2,
         )
     else:
-        return _read_outputs_v4(group)
+        return _read_outputs_v4(group, struct, redshift, inputs)
```

Comment:
At the moment, if there's a version mismatch, we don't even attempt the read, is this correct?

Reply:
I think in this case we should warn and attempt to read. Especially since this is for *newer* lightcones, not older. The warning into possible error is probably better than the warning into nothing

Reply:
Yes, I just noticed what you meant properly as I was adding coverage for missed lines -- the `else:` shouldn't be here! I think that was just a bug. I've removed it now. I also explicitly error on "older" boxes without a 21cmFAST-version attribute at all (this can be lifted once #543 is in).